codes/go/chapter_array_and_linkedlist/array.go

```diff
@@ -0,0 +1,83 @@
+package chapter_array_and_linkedlist
+
+import (
+	"fmt"
+	"math/rand"
+)
+
+// ExpectSize 预期大小。 在 Go 中，声明数组长度必须是常量表达式，
+// 所以这里我们约定，扩展后的长度为 6
+const ExpectSize = 6
```

Comment:
We use an enlarge argument.

Reply:
"The length is part of the array's type and must be a constant
expression that evaluates to a non-negative integer value." enlarge argument is
variable, not a constant expression.
https://go.dev/ref/spec#Array_types

Reply:
Please see my comments below https://github.com/krahets/hello-algo/pull/178#issuecomment-1365704881